lib/db-io.lisp

```diff
@@ -1498,54 +1526,62 @@ satisfy the optional predicate PREDICATE."
       (setf (%get-unsigned-byte ptr i) b))))
 
 (defun db-write-byte-list (cdbm keyname bytes &optional verbatim)
-  (let* ((len (length bytes)))
-    (%stack-block ((p len))
-      (save-byte-list p bytes)
-      (rletZ ((contents :cdb-datum)
-	      (key :cdb-datum))
-        (let* ((foreign-name
-		(if verbatim
-		  keyname
-		  (unescape-foreign-name keyname))))
-	  (with-cstrs ((keystring foreign-name))
-	    (setf (pref contents :cdb-datum.data) p
-		  (pref contents :cdb-datum.size) len
-		  (pref key :cdb-datum.data) keystring
-		  (pref key :cdb-datum.size) (length foreign-name))
-	    (cdbm-put cdbm key contents)))))))
+  (catch 'ignore-type
+    (let* ((len (length bytes)))
+      (%stack-block ((p len))
+        (save-byte-list p bytes)
+        (rletZ ((contents :cdb-datum)
+                (key :cdb-datum))
+          (let* ((foreign-name
+                  (if verbatim
+                    keyname
+                    (unescape-foreign-name keyname))))
+            (with-cstrs ((keystring foreign-name))
+              (setf (pref contents :cdb-datum.data) p
+                    (pref contents :cdb-datum.size) len
+                    (pref key :cdb-datum.data) keystring
+                    (pref key :cdb-datum.size) (length foreign-name))
+              (cdbm-put cdbm key contents))))))))
 
 (defun save-ffi-function (cdbm fun)
-  (let* ((encoding (encode-ffi-function fun)))
-    (db-write-byte-list cdbm
-			(ffi-function-string fun)
-			encoding
-			t)))
+  (catch 'ignore-type
```

Comment:
This seems superfluous. `DB-WRITE-BYTE-LIST` already has a `catch 'ignore-type`; do you expect that `encode-ffi-function` is going to `throw` as well in some way?